2020-10-28 10:08:16,952 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 10:08:35,939 ERROR: Exception on /start_test [GET] [in c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py", line 99, in start_test
    logout()
NameError: name 'logout' is not defined
2020-10-28 10:12:44,993 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 10:12:53,917 INFO: edit profile user name test [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:46]
2020-10-28 10:57:50,226 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 10:57:57,597 INFO: load question 318: 19 + 9 = at 2020-10-28 10:57:57.597292  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:141]
2020-10-28 10:59:44,083 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 10:59:50,205 INFO: load question 318: 19 + 9 = at 2020-10-28 10:59:50.205023  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:141]
2020-10-28 10:59:58,675 INFO: load question 318: 19 + 9 = at 2020-10-28 10:59:58.675358  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:141]
2020-10-28 10:59:58,676 INFO: question 318 end time 2020-10-28 10:59:58.676357. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:156]
2020-10-28 10:59:58,832 INFO: load question 264: 14 + 10 = at 2020-10-28 10:59:58.832295  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:141]
2020-10-28 11:12:23,873 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 11:12:33,181 INFO: load question 264: 14 + 10 = at 2020-10-28 11:12:33.181408  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:143]
2020-10-28 11:12:38,304 INFO: load question 264: 14 + 10 = at 2020-10-28 11:12:38.304089  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:143]
2020-10-28 11:12:42,526 INFO: load question 264: 14 + 10 = at 2020-10-28 11:12:42.526290  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:143]
2020-10-28 11:12:42,527 INFO: question 264 start 2020-10-28 11:12:42.527287 end time 2020-10-28 11:12:42.527287. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:162]
2020-10-28 11:15:54,137 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 11:15:57,888 INFO: load question 127: 6 + 6 = at 2020-10-28 11:15:57.888930  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:144]
2020-10-28 11:15:57,889 INFO: qStartDict {127: datetime.datetime(2020, 10, 28, 11, 15, 57, 889929)}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:148]
2020-10-28 11:16:13,046 INFO: load question 127: 6 + 6 = at 2020-10-28 11:16:13.046009  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:144]
2020-10-28 11:16:13,046 INFO: qStartDict {127: datetime.datetime(2020, 10, 28, 11, 16, 13, 46009)}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:148]
2020-10-28 11:16:13,053 INFO: question 127 start 2020-10-28 11:16:13.046009 end time 2020-10-28 11:16:13.053990. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:164]
2020-10-28 11:16:13,179 INFO: load question 114: 5 + 13 = at 2020-10-28 11:16:13.179435  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:144]
2020-10-28 11:16:13,180 INFO: qStartDict {114: datetime.datetime(2020, 10, 28, 11, 16, 13, 180431)}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:148]
2020-10-28 11:18:02,924 INFO: load question 114: 5 + 13 = at 2020-10-28 11:18:02.924364  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:144]
2020-10-28 11:18:02,924 INFO: qStartDict {114: datetime.datetime(2020, 10, 28, 11, 18, 2, 924364)}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:148]
2020-10-28 11:18:02,932 INFO: question 114 start 2020-10-28 11:18:02.924364 end time 2020-10-28 11:18:02.932343. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:164]
2020-10-28 11:23:03,890 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 11:24:17,850 INFO: load question 184: 9 + 3 = at 2020-10-28 11:24:17.850576  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:146]
2020-10-28 11:24:17,852 INFO: qStartDict {184: datetime.datetime(2020, 10, 28, 11, 24, 17, 852576)}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:150]
2020-10-28 11:24:43,347 INFO: load question 184: 9 + 3 = at 2020-10-28 11:24:43.347572  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:146]
2020-10-28 11:24:43,366 ERROR: Exception on /start_test [POST] [in c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\44792\appdata\local\programs\python\python38\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py", line 148, in start_test
    if qid not in qStartDict:
UnboundLocalError: local variable 'qStartDict' referenced before assignment
2020-10-28 11:50:23,705 INFO: create_app started. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\__init__.py:51]
2020-10-28 11:50:35,125 INFO: load question 184: 9 + 3 = at 2020-10-28 11:50:35.125097  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:144]
2020-10-28 11:50:35,125 INFO: session["questionstarttime"] {}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:147]
2020-10-28 11:50:35,126 INFO: qStartDict {184: datetime.datetime(2020, 10, 28, 11, 50, 35, 126093)} saved to session [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:152]
2020-10-28 11:50:53,717 INFO: load question 184: 9 + 3 = at 2020-10-28 11:50:53.716906  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:144]
2020-10-28 11:50:53,733 INFO: session["questionstarttime"] {'184': datetime.datetime(2020, 10, 28, 11, 50, 35)}  [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:147]
2020-10-28 11:50:53,733 INFO: qStartDict {'184': datetime.datetime(2020, 10, 28, 11, 50, 35), 184: datetime.datetime(2020, 10, 28, 11, 50, 53, 733857)} saved to session [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:152]
2020-10-28 11:50:53,734 INFO: question 184 start 2020-10-28 11:50:53.733857 end time 2020-10-28 11:50:53.734853. [in C:\Users\44792\Documents\PythonProjects\AdaptiveMath\web\app\main\routes.py:172]
